src/Aspire.Dashboard/Components/Pages/Traces.razor.cs

```diff
@@ -117,18 +117,28 @@ private async ValueTask<GridItemsProviderResult<OtlpTrace>> GetData(GridItemsPro
         TracesViewModel.Count = request.Count ?? DashboardUIHelpers.DefaultDataGridResultCount;
         var traces = TracesViewModel.GetTraces();
 
-        if (DashboardOptions.Value.TelemetryLimits.MaxTraceCount == traces.TotalItemCount && !TelemetryRepository.HasDisplayedMaxTraceLimitMessage)
+        if (traces.IsFull && !TelemetryRepository.HasDisplayedMaxTraceLimitMessage)
         {
-            await MessageService.ShowMessageBarAsync(options =>
+            TelemetryRepository.MaxTraceLimitMessage = await MessageService.ShowMessageBarAsync(options =>
             {
                 options.Title = Loc[nameof(Dashboard.Resources.Traces.MessageExceededLimitTitle)];
```

Comment:
Not fully related but why do we duplicate message bar creation logic between here and structured logs? is it appropriate to pull that out?